go/libkb/sig_chain.go

```diff
@@ -1052,6 +1052,12 @@ func (sc *SigChain) GetLinkFromSeqno(seqno keybase1.Seqno) *ChainLink {
 
 func (sc *SigChain) GetLinkFromSigID(id keybase1.SigID) *ChainLink {
 	for _, link := range sc.chainLinks {
+		if len(link.GetSigID()) == 0 {
+			// sigID might not be set for stubbed links on other users. If you're looking
+			// for a specific link by sigID of another user (e.g. for web-of-trust), then
+			// an intermediate stubbed link might otherwise cause a panic if it's not skipped.
+			continue
+		}
```

Comment:
the next line of this function goes into a flow that compares byte slices, and it will panic if either of the sigIDs it's comparing is blank. unfortunately, it is possible for a sigID to be blank on a link. if the link is on another user, and it's stubbed, then it will be empty. so far, it seems like all flows into here are looking for links on self. i think web-of-trust is the only (first?) place where the client needs to look up a chain link of another user by sig_id.